coalib/settings/SectionFilling.py

```diff
@@ -25,17 +26,38 @@ def fill_settings(sections, acquire_settings, log_printer):
     local_bears = {}
     global_bears = {}
 
+    bears_by_section = {}
+    bear_dirs_by_section = {}
+
     for section_name, section in sections.items():
-        bear_dirs = section.bear_dirs()
-        bears = list(section.get("bears", ""))
-        section_local_bears, section_global_bears = collect_bears(
-            bear_dirs,
-            bears,
-            [BEAR_KIND.LOCAL, BEAR_KIND.GLOBAL],
-            log_printer)
-        all_bears = copy.deepcopy(section_local_bears)
-        all_bears.extend(section_global_bears)
-        fill_section(section, acquire_settings, log_printer, all_bears)
+        bear_dirs_by_section[section_name] = section.bear_dirs()
+        bears_by_section[section_name] = list(section.get("bears", ""))
+
+    bears_to_collect = set(chain.from_iterable(bears_by_section.values()))
+    bear_dirs_to_collect = set(chain(*bear_dirs_by_section.values()))
+
+    all_local_bears, all_global_bears = collect_bears(
+        bear_dirs_to_collect,
```

Comment:
I don't follow.
We should be only importing each bear once. A bear is not defined by only the class of the bear, it is defined by the whole path of the file too.
So, we should "cache" this somehow, so that the warning is not thrown and the import is not happening multiple times.